### ISSUE TITLE
NSDataDetector

### ISSUE THREAD

**Issue by rlaferla** [NSHipster/articles#413]
I'd like to see an example where NSDataDetector is used to convert a string (html) without links to a string (html) with links.  It's easy to just detect and print a list but not so easy to do a search/replace...

e.g.

INPUT string

Here is a link:  http://www.google.com

OUTPUT string

Here is a link:  <a href="http://www.google.com">http://www.google.com</a>

**Comment by natecook1000** [comment]
That's a great idea for an example—thanks @rlaferla!

**Comment by mattt** [comment]
I recently updated [the `NSDataDetector` article](https://nshipster.com/nsdatadetector/). In the process, I considered adding in a section about this, but found that the topic of working with `NSAttributedString` took focus away from the subject of the article.

Some years ago, I created [TTTAttributedLabel](https://github.com/TTTAttributedLabel/TTTAttributedLabel), which offers built-in support for linking to detected data. Feel free to either use it directly or look at its implementation details to better understand how to accomplish this.
